presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/PrestoUpgrader.java

```diff
@@ -67,10 +72,42 @@ private static void upgradeUsingRpm(String pathToRpm, boolean checkDependencies)
     {
         checkRpmPackage(pathToRpm);
         String nodeps = checkDependencies ? "" : "--nodeps";
+        String tempConfigTar = gatherConfigDir();
         int upgradeRpm = executeCommand(SUBPROCESS_TIMEOUT, "sudo", "rpm", "-U", nodeps, pathToRpm);
         if (upgradeRpm != 0) {
             throw new PrestoManagerException("Failed to upgrade presto", upgradeRpm);
         }
+        deployConfigDir(tempConfigTar);
         LOGGER.debug("Successfully upgraded presto");
     }
+
+    private static String gatherConfigDir()
+            throws PrestoManagerException
+    {
+        Path tempConfigTar;
+        try {
+            tempConfigTar = Files.createTempFile("presto_config", ".tar.gz");
+        }
+        catch (IOException e) {
+            throw new PrestoManagerException("Failed to create a temp tar file", e);
+        }
+        int tarConfigDir = executeCommand("tar", "-czv", "-f", tempConfigTar.toString(), "-C", CONFIG_DIR.toString(), ".");
```

Comment:
There's no reason to `tar` the files here.

And you aren't using your temporary file; you're creating a new file with the same name in your current directory.

`tar -czv -f tempFile configDir` would be better, if you need to use `tar`.